server/planning/locations/locations_service_async.py

```diff
@@ -0,0 +1,40 @@
+from typing import Any
+
+from planning.core.service import BasePlanningAsyncService
+from planning.events.events_service import EventsAsyncService
+from planning.types import LocationResourceModel
+
+from superdesk.core.types import SearchRequest
+from superdesk.core.utils import generate_guid, GUID_NEWSML
+from superdesk.resource_fields import ID_FIELD
+
+
+class LocationsAsyncService(BasePlanningAsyncService[LocationResourceModel]):
+    async def on_create(self, docs: list[LocationResourceModel]) -> None:
+        await super().on_create(docs)
+
+        for doc in docs:
+            if not doc.guid:
+                doc.guid = generate_guid(type=GUID_NEWSML)
```

Comment:
Seems like a location should ALWAYS have a `guid` value. In this case I would use Pydantic to do this for us, and we can remove this `on_create` entirely from this service.

On the model you would replace
```python
guid: str | None = None
```
with
```python
guid: Annotated[str, Field(default_factory=lambda: generate_guid(type=GUID_NEWSML))]
```

And this way Pydantic will ensure that the `guid` field will always have a value, and never be `None`